eventsource-client/src/error.rs

```diff
@@ -11,8 +13,18 @@ impl ResponseWrapper {
     pub fn status(&self) -> u16 {
         self.response.status().as_u16()
     }
-    pub fn headers(&self) -> &hyper::header::HeaderMap {
-        self.response.headers()
+    pub fn headers(&self) -> Result<HashMap<&str, &str>> {
+        let headers = self.response.headers();
+        let mut map = HashMap::new();
+        for (key, value) in headers.iter() {
+            let key = key.as_str();
+            let value = match value.to_str() {
+                Ok(value) => value,
+                Err(err) => return Err(Error::InvalidParameter(Box::new(err))),
```

Comment:
This error is still hyper specific no?

Maybe get(&str) -> Result<String> to make this more straightforward?

Reply:
👍

Reply:
> This error is still hyper specific no?

It's eventsource specific

Reply:
it's not a hyper error

Reply:
> Maybe get(&str) -> Result to make this more straightforward?

Makes the thing kind of painful to use IMHO